Tests/SentryTests/Helper/TestSentryNSProcessInfoWrapper.swift

```diff
@@ -0,0 +1,32 @@
+import Sentry
+
+class TestSentryNSProcessInfoWrapper: SentryNSProcessInfoWrapper {
+    struct Override {
+        var isLowPowerModeEnabled: Bool?
+        var thermalState: ProcessInfo.ThermalState?
+        var processorCount: UInt?
+    }
+
+    var overrides = Override()
+
+    override var thermalState: ProcessInfo.ThermalState {
+        overrides.thermalState ?? super.thermalState
+    }
+
+    override var isLowPowerModeEnabled: Bool {
+        overrides.isLowPowerModeEnabled ?? super.isLowPowerModeEnabled
+    }
+
+    override var processorCount: UInt {
+        overrides.processorCount ?? super.processorCount
+    }
+
+    @available(iOS 9.0, macOS 12.0, *)
+    func sendPowerStateChangeNotification() {
+        SentryDependencyContainer.sharedInstance().notificationCenterWrapper.postNotificationName(NSNotification.Name.NSProcessInfoPowerStateDidChange, object: ProcessInfo.processInfo)
+    }
+
+    func sendThermalStateChangeNotification() {
+        SentryDependencyContainer.sharedInstance().notificationCenterWrapper.postNotificationName(ProcessInfo.thermalStateDidChangeNotification, object: ProcessInfo.processInfo)
+    }
```

Comment:
`m`: We had troubles with flaky tests in the past when using the notification center. Please consider using `TestNSNotificationCenterWrapper` here.

Reply:
Oh, I have everything using the wrapper, but I think I forgot to swap it out for its test subclass.

On a related not, we might think about refactoring how the notification center wrapper is passed around. We have a singleton instance of `SentryDependencyContainer`, but pass the reference to its notificationCenterWrapper to other objects' init methods. We could instead just swap out the `SentryDependencyContainer` singleton's object and always call `SentryDependencyContainer.sharedInstance.notificationCenterWrapper` from those objects. What do you think?

Reply:
Hmm, reading through TestSentryNSNotificationCenterWrapper, I'm not sure I understand how it helps with the test issues we were seeing. It doesn't wrap any posting or receiving of notifications, and still just calls through to the default notification center for all of its calls anyways. What is its purpose?

When I substituted in the TestNSNotificationCenter, the test that used to pass now fails with an EXC_BAD_ACCESS in `Invocations.record`.

Reply:
> and still just calls through to the default notification center for all of its calls anyways

I fixed that with https://github.com/getsentry/sentry-cocoa/pull/2554. Please merge 8.0.0 into your branch.

Reply:
That actually moves away from the direction I'm suggesting. After your changes, the only thing `TestSentryNSNotificationCenterWrapper` does is record invocations of observer registrations. When the SDK expects to receive a notification, where does the actual call come from to test? Can you link me to the line of code that forwards such a call?

Reply:
We use `Dynamic(sut).methodTheNotificationCenterShouldCall` instead of keeping track of observers. It's a bit hacky, but pragmatic.

https://github.com/getsentry/sentry-cocoa/blob/75ba8bdd4f75a298894b3725cca7806f20e8b626/Tests/SentryTests/Integrations/Session/SentrySessionTrackerTests.swift#L366-L380